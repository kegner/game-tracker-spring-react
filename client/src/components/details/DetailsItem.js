import React from 'react'

const DetailsItem = (props) => {

  const { label, value, currency } = props;

  const formatter = new Intl.NumberFormat("en-us", {
    style: "currency",
    currency: "USD",
  });

  let formattedValue = value;
  if (currency && value) {
    formattedValue = formatter.format(value);
  }

  return (
    <>
      <strong>{label}</strong><br />
      <span>{formattedValue}</span>
    </>
  )
}

export default DetailsItem
