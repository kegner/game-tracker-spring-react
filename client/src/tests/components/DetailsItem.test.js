import React from 'react';
import { render, getByText, queryByText } from '@testing-library/react';
import DetailsItem from '../../components/details/DetailsItem';

describe("DetailsItem", () => {

  it("exists", () => {
    const { asFragment } = render(
      <DetailsItem />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("shows normal values", () => {
    const { container } = render(
      <DetailsItem label="My item" value={12.34} currency={false} />
    );

    expect(getByText(container, "My item")).toBeVisible();
    expect(queryByText(container, "$12.34")).toBe(null);
    expect(getByText(container, "12.34")).toBeVisible();
  });

  it("shows currency values", () => {
    const { container } = render(
      <DetailsItem label="My item" value={12.34} currency={true} />
    );

    expect(getByText(container, "My item")).toBeVisible();
    expect(getByText(container, "$12.34")).toBeVisible();
  });

});