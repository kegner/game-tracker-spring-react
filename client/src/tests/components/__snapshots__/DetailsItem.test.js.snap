// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsItem exists 1`] = `
<DocumentFragment>
  <strong />
  <br />
  <span />
</DocumentFragment>
`;
